src/main/java/touro/peg/BoardSolutionChecker.java

```diff
@@ -0,0 +1,84 @@
+package touro.peg;
+
+import touro.peg.TriangleBoard;
+import touro.tree.TriangleBoardTree;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BoardSolutionChecker
+{
+    private TriangleBoardTree tree;
+    public BoardSolutionChecker(TriangleBoard board)
+    {
+        this.tree = new TriangleBoardTree(board);
+    }
+
+    public List<TriangleBoard> getSolutions()
+    {
+        ArrayList<TriangleBoard> solutions = new ArrayList<>();
+        if (tree == null)
+        {
+            return solutions;
+        }
```

Comment:
Why would tree be null?

Reply:
Oh this came from when I was instantiating the tree in the Board - but not in the copyBoard constructor.
Would it be a similar problem since I couldn't instantiate the solutions class within the copy board constructor?